### ISSUE TITLE
Auto-detect mobile hotspots using a bounding box, and filter them out

### ISSUE THREAD

**Issue by crankycoder** [mozilla/MozStumbler#1106]
We currently use a naive approach to detecting mobile hotspots.

In most cases, these are going to be tethered cell phones, and in some cases you'll see things like in-car hotspots, trains or buses.

On the device, we already capture the BSSID and lat/long during a stumble.  We can compute a bounding box for the min lat/long and the max lat/long for each BSSID.

Outdoors, the typical range for a wifi signal is going to be about 100m.  So let's say that if we detect that the bounding box has a distance of 1km or more - that's going to be a mobile hotspot.

This should be safe to use even with long range wifi because:
- long range wifi is pretty rare
- long range wifi is directional
- the ichnaea server only gives location fixes when you have > 2 (possibly 3?) BSSIDs submitted

We can run the computation for mobile hotspots at the time we do an upload since we're going to be reading back the report logs anyway.

**Comment by Djfe** [comment]
Nice feature that could reduce data usage

An issue that comes to my mind is that this might unhook the server-side blacklist for moving wifis

Possible Solutions:
1. The client should still upload a list of wifis that have been blocked by the client so that the server can add them to the list and block these wifis for older database entries and new uploaded ones (by other clients, that don't have such a feature)
This would need an api extention.
2. Upload only two of the reports that are few enough away from each other for the server-side logic to detect that they are moving and delete all other reports

**Comment by garvankeeley** [comment]
@hannosch Does ichnaea handle mobile hotspot detection in some fashion?

**Comment by hannosch** [comment]
ichnaea does pretty much the same here as Victor suggests. It continually updates the aggregate wifi position based on incoming observations. If those observations tell it that the wifi has moved more than 5km away from the last location, the entry gets removed and put on a blacklist. After a week we allow entries to be added again for the same wifi, so if it has moved and is now in a new stable position, we accept it again. If it moves yet again, it gets again removed/blacklisted. If something does this dance at least a couple of times during consecutive weeks it gets permanently blacklisted.

This logic isn't too smart, but currently has to deal with the fact that we have no way to throw away or identify single bad observations. So we don't want to permanently blacklist something, just because we got one bad observation.

**Comment by cascheberg** [comment]
I got the impression that server-side filtering is more effective and should include all raw client data. Referencing #867 as a side note.

**Comment by crankycoder** [comment]
Given https://github.com/mozilla/MozStumbler/issues/1106#issuecomment-62749905 , I propose that we stop adding to the wifi blacklist on the client side.  Thoughts @garvankeeley ?

**Comment by crankycoder** [comment]
Marking this as won't fix since the server side does this already. No need to re-implement this.
